CoreScriptsRoot/Modules/SocialUtil.lua

```diff
@@ -0,0 +1,124 @@
+--[[
+	// FileName:    SocialUtil.lua
+	// Written by:  TheGamer101
+	// Description: Utility code related to social features.
+]]
+local SocialUtil = {}
+
+--[[ Services ]]--
+local Players = game:GetService("Players")
+
+--[[ Constants ]]--
+local THUMBNAIL_SIZE_MAP = {
+	[Enum.ThumbnailSize.Size48x48]   =  48,
+	[Enum.ThumbnailSize.Size180x180] = 180,
+	[Enum.ThumbnailSize.Size420x420] = 420
+}
+
+local THUMBNAIL_FALLBACK_URLS = {
+	[Enum.ThumbnailType.HeadShot] = "https://www.roblox.com/headshot-thumbnail/image?width=%d&height=%d&format=png&userId=%d",
+	[Enum.ThumbnailType.AvatarBust] = "https://www.roblox.com/bust-thumbnail/image?width=%d&height=%d&format=png&userId=%d",
+	[Enum.ThumbnailType.AvatarThumbnail] = "https://www.roblox.com/avatar-thumbnail/image?width=%d&height=%d&format=png&userId=%d"
+}
+
+local GET_PLAYER_IMAGE_DEFAULT_TIMEOUT = 5
+local DEFAULT_THUMBNAIL_SIZE = Enum.ThumbnailSize.Size48x48
+local DEFAULT_THUMBNAIL_TYPE = Enum.ThumbnailType.AvatarThumbnail
+local GET_USER_THUMBNAIL_ASYNC_RETRY_TIME = 1
+
+--[[ Utility ]]--
+local function CreateSignal()
+	local sig = {}
+
+	local mSignaler = Instance.new('BindableEvent')
+
+	local mArgData = nil
+	local mArgDataCount = nil
+
+	function sig:fire(...)
+		mArgData = {...}
+		mArgDataCount = select('#', ...)
+		mSignaler:Fire()
+	end
+
+	function sig:connect(f)
+		if not f then error("connect(nil)", 2) end
+		return mSignaler.Event:Connect(function()
+			f(unpack(mArgData, 1, mArgDataCount))
+		end)
+	end
+
+	function sig:wait()
+		mSignaler.Event:wait()
+		if not mArgData then
+			error("Missing arg data, likely due to :TweenSize/Position corrupting threadrefs.")
+		end
+		return unpack(mArgData, 1, mArgDataCount)
+	end
+
+	return sig
+end
+
+--[[ Functions ]]--
+
+-- The thumbanil isn't guaranteed to be generated, this will just create the url using string.format and immediately return it.
+function SocialUtil.GetFallbackPlayerImageUrl(userId, thumbnailSize, thumbnailType)
+	local sizeNumber = THUMBNAIL_SIZE_MAP[thumbnailSize]
```

Comment:
This function will error if thumbnailSize is nil. This is inconsistent with GetPlayerImage which uses the default thumbnail size in the case that it is nil. Should this function do that instead?